### PR TITLE
Add `wg-const-eval` team

### PR DESCRIPTION
**Pull request by ecstatic-morse**
...with @oli-obk and @RalfJung as co-leads and myself as a member. I don't think this implies any additional level of commitment from the two leads, it's just a reflection of seniority.

We might need to block this on me reviving rust-lang/compiler-team#201? Not sure what the process is here.

r? @XAMPPRocky 
(since they requested this on Zulip)

### CODE REVIEW

File: teams/wg-const-eval.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+leads = ["oli-obk", "RalfJung"]
+members = [
+    "ecstatic-morse",
+    "oli-obk",
+    "RalfJung",
+]
+
+[website]
+name = "Compile-time function evaluation"
+description = "Expanding the capabilities of compile-time function evaluation in Rust"
```
Comment (RalfJung) [inline teams/wg-const-eval.toml:15]: "and ensuring their soundness" or so? I do not entirely see my work reflected in this description. ;)

**Comment by RalfJung** [comment]
Fine for me, thanks for taking initiative! Since I am not very familiar with some aspects of const-eval, in particular the static checks, I'd also be fine with @oli-obk as the sole lead. My involvement here is mainly through Miri, and to lend my theoretical and PL research experience for matters of const-soundness (as reflected in our dynamic checks). But the further we move from the Miri engine to the rest of rustc, the less familiar I am with the territory. ;)

But maybe leads don't have to know everything, then that's okay with me as well. :D

File: teams/wg-const-eval.toml (hunk 1 above)
Comment (ecstatic-morse) [reply teams/wg-const-eval.toml:15]: Would you settle for an adverb? "Soundly expanding"?

Comment (RalfJung) [reply teams/wg-const-eval.toml:15]: That sounds good, too. :D

File: teams/wg-const-eval.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+[people]
+leads = ["oli-obk", "RalfJung"]
+members = [
+    "ecstatic-morse",
+    "oli-obk",
+    "RalfJung",
+]
+
+[website]
+name = "Compile-time function evaluation"
```
Comment (XAMPPRocky) [inline teams/wg-const-eval.toml:14]: This should include "Working Group" or the appropriate moniker.

**Comment by XAMPPRocky** [comment]
@ecstatic-morse LGTM! Thanks for making the PR, if the group needs or wants shared permissions on repos such as miri you can add the following. I can't merge the PR (only core can merge on `team`).

```toml
[permission]
bors.miri.review = true
```

**Comment by RalfJung** [comment]
There's a dedicated Miri team/WG already, I don't think const-eval needs any extra permissions (well maybe r+ for rustc itself but we all already have that from elsewhere).

**Comment by oli-obk** [comment]
Yea, miri is independent from const eval. While miri-engine decisions made for const eval affect miri, that's like any teams decisions having effects on other teams, so that's fine.

**Comment by RalfJung** [comment]
Also the Miri and cosnt-eval teams have good communication as they are 66% the same people. ;)

**Comment by ecstatic-morse** [comment]
I updated the title @XAMPPRocky. [rust-lang/const-eval](https://www.github.com/rust-lang/const-eval) doesn't use bors (there's no rust code), and that's the only repo that's managed by this working group.

File: teams/wg-const-eval.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "RalfJung",
+]
+
+[website]
+name = "Compile-time Function Evaluation Working Group"
+description = "Soundly expanding the capabilities of compile-time function evaluation in Rust"
+repo = "https://github.com/rust-lang/const-eval"
+
+[github]
+orgs = ["rust-lang", "rust-lang-nursery"]
```
Comment (RalfJung) [inline teams/wg-const-eval.toml:19]: Is there any reason to have this team on the nursery?

**Comment by ecstatic-morse** [comment]
Closing due to inactivity. I was explicitly asked to do this, so I'm not sure what the deal is.

**Comment by RalfJung** [comment]
No need to be snappy, probably it just dropped off of someone's radar. Just ping someone who can hopefully push this forward. :)
There's also [an unanswered question](https://github.com/rust-lang/team/pull/332#discussion_r418785769).

@XAMPPRocky @pietroalbini what does it take to land this?

**Comment by XAMPPRocky** [comment]
The PR has been good to merge for me, I don't have merge access to team. It has to be done by a core team member.

**Comment by RalfJung** [comment]
@ecstatic-morse can you reopen the PR? I can take care of pushing it through.

File: teams/wg-const-eval.toml (hunk 3 above)
Comment (nikomatsakis) [inline teams/wg-const-eval.toml:19]:
```suggestion
orgs = ["rust-lang"]
```

**Pull request merged**